File: src/main/java/no/nav/domain/pensjon/regler/Merknad.java
package no.nav.domain.pensjon.regler;


import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

/**
 * Liste med forklaringer, unntak og avvisningsgrunner. Merknader legges i
 * resultatet av et kall til en regeltjeneste.
 */
public class Merknad implements Serializable {

    

    /**
     * Identifiserer merknaden. Navnekonvensjonen er:
     * Regelsettnavn.regelnavn.index, der ".index" er en opsjon.
     */
    private String kode;

    /**
     * Beskrivende
     */
    private List<String> argumentListe;


}

File: src/main/kotlin/no/nav/pensjon/regler/domain/Merknad.kt
package no.nav.pensjon.regler.domain

import java.io.Serializable

class Merknad : Serializable {
    /**
     * Identifiserer merknaden. Navnekonvensjonen er:
     * Regelsettnavn.regelnavn.index, der ".index" er en opsjon.
     */
    var kode: String? = null

    /**
     * Beskrivende
     */
    var argumentListe: MutableList<String> = mutableListOf()

    constructor()
    constructor(merknad: Merknad) {
        kode = merknad.kode
        argumentListe = mutableListOf<String>().apply {
            addAll(merknad.argumentListe)
        }
    }

}